balena/models/history.py

```diff
@@ -40,25 +41,26 @@ def _history_timerange_filter_with_guard(self, fromDate=None, toDate=None):
             else:
                 raise exceptions.InvalidParameter("toDate", toDate)
 
-        rawFilterString = "$filter=" + " and ".join(rawTimeRangeFilters)
+        return rawTimeRangeFilters
 
-        if len(rawTimeRangeFilters) > 0:
-            return rawFilterString
-        else:
-            return None
+    def _history_compile_raw_filter(self, **options):
+        rawOptionsFilter = []
+        for key in options:
+            rawOptionsFilter.append(Template("$key%20eq%20'$value'").safe_substitute(key=key, value=options[key]))
```

Comment:
@fisehara Just to know, why did you prefer Template + safe_substitute rather than
```suggestion
            rawOptionsFilter.append(f"{key}%20eq%20'{options[key]}'")
```
? afaik they will both achieve the same thing

Reply:
@otaviojacobi I just copy/paste the existing procedure, and agree with your shortened version.